addons/mail/static/src/composer/suggestion_hook.js

```diff
@@ -73,27 +75,54 @@ export function useSuggestion() {
             inProgress: false,
             rpcFunction: undefined,
         },
-        insert(option) {
-            const cursorPosition = comp.props.composer.selection.start;
-            const content = comp.props.composer.textInputContent;
-            let textLeft = content.substring(0, self.search.position + 1);
-            let textRight = content.substring(cursorPosition, content.length);
-            if (self.search.delimiter === ":") {
-                textLeft = content.substring(0, self.search.position);
-                textRight = content.substring(cursorPosition, content.length);
-            }
-            const recordReplacement = option.label;
+        insert(option, callback = () => {}) {
```

Comment:
We should type this `option` param; It also doesn't look like an option at all? It's more like suggestion data?